test/hotspot/jtreg/vmTestbase/nsk/jdi/AttachingConnector/attachnosuspend/attachnosuspend001.java

```diff
@@ -96,19 +94,17 @@ private int runIt(String argv[], PrintStream out) {
         long timeout = argHandler.getWaitTime() * 60 * 1000;
         attempts = (int)(timeout / delay);
 
-        port = argHandler.getTransportPort();
-        listenPort = argHandler.getTransportPort();
-
         String java = argHandler.getLaunchExecPath()
                         + " " + argHandler.getLaunchOptions();
-        String cmd = java +
-                " -Xdebug -Xnoagent -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=" +
-                listenPort + " " + DEBUGEE_CLASS;
+        String cmd = java
+                + " -Xdebug -Xnoagent -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=0"
+                + " " + DEBUGEE_CLASS;
```

Comment:
You've lost logging of the port you end up listening on. Is the dynamically discovered listening port logged somewhere else?

Reply:
DebugeeProcess.redirectOutputAndDetectListeningAddress logs it
(see DebugeeProcess.java, line 428)
Also IORedirector logs debugee output (including "Listening for transport ... at address ..." line)